src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -673,7 +683,104 @@ else if (v instanceof EntityValue)
             });
             return (cc, tt) -> result[0];
         });
+        // =======================================
+        expression.addContextFunction("structure_block_save", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerWorld lv2 = cc.s.getWorld();
+
+            StructureManager lv3 = lv2.getStructureManager();
+            Structure lv6;
+            String name;
+
+            name = lv.get(0).getString();
+            Identifier struident = Identifier.tryParse(name);
+            if (name.isEmpty()||name.endsWith(":")||struident == null) {
+                return Value.NULL;
+            }
+            lv6 = lv3.getStructureOrBlank(struident);
+
+            // name,start,dimensions, ignoreEntities, ignoredBlock,~~author~~,disk
+            lv6.saveFromWorld(lv2,
+                    new BlockPos(((NumericValue) lv.get(1)).getInt(), ((NumericValue) lv.get(2)).getInt(),
+                            ((NumericValue) lv.get(3)).getInt()),
+                    new Vec3i((int) lv.get(4).readInteger(), (int) lv.get(5).readInteger(),
+                            (int) lv.get(6).readInteger()),
```

Comment:
There is a specific way to get triplets of positions from command arguments, so that users can either input 3 numbers or a list of 3 numbers. See code for block() function and look for BlockLocator class

Reply:
done.